middleware/supervisor/detectors-supervisor.tf

```diff
@@ -0,0 +1,47 @@
+resource "signalfx_detector" "heartbeat" {
+  name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] Supervisor heartbeat"
+
+  program_text = <<-EOF
+		from signalfx.detectors.not_reporting import not_reporting
+		signal = data('supervisor.state', filter=${module.filter-tags.filter_custom}).publish('signal')
+		not_reporting.detector(stream=signal, resource_identifier=['host'], duration='${var.heartbeat_timeframe}').publish('CRIT')
+  EOF
+
+  rule {
+    description           = "has not reported in ${var.heartbeat_timeframe}"
+    severity              = "Critical"
+    detect_label          = "CRIT"
+    disabled              = coalesce(var.heartbeat_disabled, var.detectors_disabled)
+    notifications         = coalescelist(var.heartbeat_notifications, var.notifications)
+    parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} on {{{dimensions}}}"
+  }
+}
+
+resource "signalfx_detector" "process_state" {
+  name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] Supervisor process"
+
+  program_text = <<-EOF
+		signal = data('supervisor.state', filter=${module.filter-tags.filter_custom})${var. process_state_aggregation_function}.${var. process_state_transformation_function}(over='${var.process_state_transformation_window}').publish('signal')
+		detect(when(signal > ${var.process_state_threshold_critical})).publish('CRIT')
```

Comment:
this way could lead to undesired alerts no ?

for example STOPPING or STARTING states will trigger alerts which seem not legitimate.

of course we could hope process_state_transformation_window is big enough to avoid that but it seems to be a better approach to use custom detector and alerts only if one of undesired state : https://docs.signalfx.com/en/latest/detect-alert/alert-condition-reference/custom-threshold.html

I suggest to map:

- STARTING and STARTED as OK
- UNKNOWN and STOPPING as warning  
- every other states as critical

Reply:
As discussed I think we can merge it that way and we will improve it if needed later.